src/LieTheory/WeylGroup.jl

```diff
@@ -926,3 +926,117 @@ function Base.iterate(iter::WeylOrbitIterator, state::WeylIteratorNoCopyState)
   (wt, _), state = it
   return deepcopy(wt), state
 end
+
+###############################################################################
+# Misc other things
+
+@doc raw"""
+    geometric_representation(W::WeylGroup) -> MatrixGroup{ZZRingElem}, Map{WeylGroup, MatrixGroup{ZZRingElem}}
+
+Return the geometric representation `G` of the Weyl group `W`,
+together with the isomorphism `hom` from `W` to `G`.
+
+This representation is defined by `coefficients(a) * hom(x) == coefficients(a * x)`
+for all `x` in `W` and `a` a simple root of `root_system(W)`. By linear extension,
+this also holds for all elements `a` in the root space of `root_system(W)`.
+
+See [Hum90; Sect. 5.3](@cite) for more details.
+
+# Examples
+```jldoctest
+julia> R = root_system(:B, 3); W = weyl_group(R);
+
+julia> r = positive_root(R, 8)
+a_1 + a_2 + 2*a_3
+
+julia> x = W([1, 2, 1, 3])
+s1 * s2 * s1 * s3
+
+julia> G, hom = geometric_representation(W)
+(Matrix group of degree 3 over ZZ, Map: W -> G)
+
+julia> coefficients(r) * hom(x)
+[1   1   0]
+
+julia> coefficients(r * x)
+[1   1   0]
+```
+"""
+@attr Tuple{
+  MatrixGroup{ZZRingElem,ZZMatrix},MapFromFunc{WeylGroup,MatrixGroup{ZZRingElem,ZZMatrix}}
```

Comment:
Maybe we should define an alias `const ZZMatrixGroup = MatrixGroup{ZZRingElem,ZZMatrix}` ?

Reply:
do we need `matrix_group_type(R)`?

Reply:
I just did both of these. Please someone have a look. (and also fixed some docstrings while looking for applications of this new function)